lib/sql_parser.mly

```diff
@@ -170,6 +172,7 @@ statement: CREATE ioption(temporary) TABLE ioption(if_not_exists) name=table_nam
                 Function.add (List.length params) (Ret Any) name.tn; (* FIXME void *)
                 CreateRoutine (name, None, params)
               }
+         | is_recursive=cte_with ctes=commas(cte) stmt=select_stmt { Cte_select { ctes; stmt; is_recursive }}     
```

Comment:
move next to select_stmt
but actually this is suspicious, i would expect cte to be part of select_stmt, not a separate branch (iow no separate Cte_select ctor)

Reply:
Yes, since this is first step I didn't think a lot. But now I realized:

| Statement  | Query                                      |
|-----------|---------------------------------------------------------------|
| **SELECT** | ```WITH cte AS (SELECT col_1, col_2 FROM table_1 WHERE col_3 > 100) SELECT * FROM cte;``` |
| **INSERT** | ```WITH cte AS (SELECT 'value_1', 12345) INSERT INTO table_1 (col_1, col_2) SELECT * FROM cte;``` |
| **UPDATE** | ```WITH cte AS (SELECT col_1, col_2 FROM table_1 WHERE col_3 = 'value') UPDATE table_1 SET col_2 = col_2 * 1.1 WHERE col_1 IN (SELECT col_1 FROM cte);``` |
| **DELETE** | ```WITH cte AS (SELECT col_1 FROM table_1 WHERE col_4 < '2023-01-01') DELETE FROM table_1 WHERE col_1 IN (SELECT col_1 FROM cte);``` |

Reply:
Done. 
Got rid of `Cte_select` at all, and made ctes a part of `SELECT`. 
And `INSERT SELECT` + cte started work simply by the fact that  they fell into place

Reply:
So the current state of supporting of ctes is : 
<img width="832" alt="MySQL____MySQL_8_4_Reference_Manual____15_2_20_WITH__Common_Table_Expressions_" src="https://github.com/user-attachments/assets/bdd0384c-d39e-42e5-a278-beb02dad2ff6">
Thanks for it!